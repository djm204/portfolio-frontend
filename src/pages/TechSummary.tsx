import React from 'react';

const TechSummary = () => {
    return (
        <React.Fragment>
            TechSummary
        </React.Fragment>
    );
}

export default TechSummary;