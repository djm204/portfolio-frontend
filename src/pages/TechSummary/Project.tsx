import React, { useState } from 'react'
import cnames from 'classnames'
import '../../stylesheets/Project.css'

type ProjectProps = {
  name: string
  link: string
  imgSrc: string
  points: Array<string> | string
  role: string
  className: string
  dataAos?: string
  dataAosDuration?: string
  dataAosEasing?: string
  dataAosOnce?: string
}

const Project = (props: ProjectProps) => {
  const [isExpanded, setIsExpanded] = useState(false)
  const { role, name, link, points, className, imgSrc, dataAosDuration, dataAosEasing } = props

  const handleClick = (event: React.MouseEvent<HTMLImageElement, MouseEvent>) => {
    if (event) event.persist()
    setIsExpanded(!isExpanded)
  }

  return (
    <div id={className} className={cnames('Project', className)}>
      <div
        onClick={handleClick}
        className={cnames('ExpandedProjectImageBackground', { '--expanded': isExpanded })}
      >
        <div className="Project__Image__Wrapper__Expanded">
          <img className="Project__Image__Expanded" alt={name} src={imgSrc} />
        </div>
      </div>
      <img onClick={handleClick} className={cnames('Project__Image')} alt={name} src={imgSrc} />

      <div className={cnames('Project__Details', { '--expanded': isExpanded })}>
        <h3
          className="Project__Name "
          data-aos-duration={dataAosDuration}
          data-aos-easing={dataAosEasing}
        >
          {name}
        </h3>
        <div className="Project__Role ">{role}</div>
        <ul className="Project__Notables ">
          {typeof points !== 'string' ? (
            points.map(point => <li key={point}>{point}</li>)
          ) : (
            <li>{points}</li>
          )}
        </ul>
        <div className="Project__Link__Wrapper">
          {link === 'NPA' ? (
            <span className="Project__NotPublic ">*Not Publicly Available</span>
          ) : (
            <a href={link} rel="noopener noreferrer" target="_blank" className="Project__Link">
              View Project
            </a>
          )}
        </div>
      </div>
    </div>
  )
}

export default Project
