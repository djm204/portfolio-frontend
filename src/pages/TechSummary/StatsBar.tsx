import React from 'react'
import '../../stylesheets/StatsBar.css'

type StatsBarProps = {
  name: string
  percentage: number
}

const StatsBar = (props: StatsBarProps) => {
  const { name, percentage } = props

  return (
    <div className="StatsBar__Wrapper">
      <div className="StatsBar__Name">{name}</div>
      <div className="StatsBar__Container">
        <div className="StatsBar" style={{ width: percentage - 10 + '%' }}>
          {percentage + '%'}
        </div>
      </div>
    </div>
  )
}

export default StatsBar
