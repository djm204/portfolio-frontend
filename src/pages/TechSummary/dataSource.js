export const projects = [
  {
    className: 'Projects_Cashier',
    name: 'Cashier',
    role: 'Developer Team Lead / Full Stack Developer',
    link: 'NPA',
    imgSrc: 'img/cashier.jpg',
    imgExpandedSrc: 'img/cashier--expanded.jpg',
    points: [
      'Pair programming / mentoring & unblocking reports.',
      'Delegation and prioritization of tasks.',
      'React/Redux App for Admin and Checkout.',
      'Hybrid PHP / Go backend.',
      '4 years on the team.',
    ],
    dataAos: 'fade-right',
    dataAosDuration: '1000',
    dataAosEasing: 'ease-in-out',
    dataAosOnce: 'true',
  },
  {
    className: 'Projects_Staples',
    name: 'Staples Project',
    link: 'https://staples.ca',
    imgSrc: 'img/staples.jpg',
    imgExpandedSrc: 'img/staples--expanded.jpg',
    role: 'Developer Team Lead / Full Stack Developer',
    points: [
      'React/Redux App.',
      "Customized Cashier to meet Staples' business requirements.",
      'Successful delivery on tight timeline.',
    ],
    dataAos: 'fade-up',
    dataAosDuration: '1000',
    dataAosEasing: 'ease-in-out',
    dataAosOnce: 'true',
  },
  {
    className: 'Projects_MultiCurrency',
    name: 'Bold MultiCurrency',
    link: 'https://boldcommerce.com/multi-currency',
    imgExpandedSrc: 'img/multicurrency--expanded.jpg',
    imgSrc: 'img/multicurrency.jpg',
    role:
      'Completed the first iteration of the frontend of the app in its entirety, pretty much solo. *cracks knuckles*',
    points: ['React/Redux App', 'Conversion engine', 'Integrated with Cashier'],
    dataAos: 'fade-up',
    dataAosDuration: '1000',
    dataAosEasing: 'ease-in-out',
    dataAosOnce: 'true',
  },
  {
    className: 'Projects_DeepRoots',
    name: 'DRSD - Website',
    link: 'https://www.deeproots-strongdance.com/',
    imgExpandedSrc: 'img/deeproots--expanded.jpg',
    imgSrc: 'img/deeproots.jpg',
    role: 'Designed and built entire site',
    points: ['React/Redux', 'API endpoint for adding to mailing list.'],
    dataAos: 'fade-left',
    dataAosDuration: '1000',
    dataAosEasing: 'ease-in-out',
    dataAosOnce: 'true',
  },
]

export const skills = [
  {
    name: 'HTML',
    percentage: 95,
  },
  {
    name: 'CSS',
    percentage: 90,
  },
  {
    name: 'JavaScript',
    percentage: 90,
  },
  {
    name: 'React',
    percentage: 90,
  },
  {
    name: 'Redux',
    percentage: 85,
  },
  {
    name: 'TypeScript',
    percentage: 75,
  },
  {
    name: 'NodeJS',
    percentage: 65,
  },
  {
    name: 'Go',
    percentage: 65,
  },
  {
    name: 'K8s',
    percentage: 55,
  },
  {
    name: 'GCP',
    percentage: 55,
  },
]
