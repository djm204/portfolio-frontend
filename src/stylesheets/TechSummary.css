.TechSummaryPage__Introduction__StatBars {
  flex-basis: 50%;
}

.TechSummaryPage__InfoSection {
  background-color: rgba(172, 197, 255, 0.4);
}

.TechSummaryPage__Projects {
  flex-direction: column;
}

.TechSummaryPage__Projects h2 {
  text-align: center;
  padding-bottom: 20px;
  padding-top: 20px;
}

.TechSummary__Projects__Wrapper {
  display: flex;
  justify-content: space-evenly;
}

.Project {
  padding: 25px;
  position: relative;
  width: 25%;
}

.Project--hidden {
  visibility: hidden;
  opacity: 0;
  transition: all 0.2s ease-in-out;
}

.Project--visible {
  visibility: visible;
  opacity: 1;
  transition: all 1s ease-in-out;
}

.Project--SlideIn {
  visibility: visible;
  transform: translateX(150px);
  animation: slideLeft 0.8s ease forwards;
}

.Project--SlideIn:nth-child(odd) {
  animation-duration: 0.6s;
}

.Project__Role {
  font-style: italic;
  transition: all 1s ease-in-out;
}

.Project__Details {
  transition: all 1s ease-in-out;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  height: 250px;
}

.Project__Image {
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  z-index: -1;
}

.Project__Notables {
  list-style: none;
}

.Project__NotPublic {
  font-weight: bold;
}

.Project__Link__Wrapper {
  flex-grow: 1;
  display: flex;
  justify-content: flex-start;
}
.Project__Link,
.Project__Link a {
  color: rgb(0, 34, 111);
}

.Project__Link {
  display: inline-block;
  padding: 10px 20px;
  border-radius: 3px;
  border: 1px solid rgba(26, 95, 255, 1);
  margin-top: auto;
}

.Project__Link:hover {
  text-decoration: none;
  color: rgba(172, 197, 255, 1);
  background-color: rgba(26, 95, 255, 1);
  border: 1px solid rgb(0, 34, 111);
  transition: background-color ease-in-out 0.2s;
}

@keyframes slideLeft {
  to {
    transform: translateX(0px);
  }
}
