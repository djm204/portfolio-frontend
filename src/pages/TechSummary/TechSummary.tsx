import React from 'react';
import { InfoSection } from '../../components/layout/InfoSection';
import Project from '../../components/Project';


const TechSummary = () => {
    const projects = [
        {
            className: 'Projects_Cashier',
            name: 'Cashier',
            role: 'Developer Team Lead / Full Stack Developer',
            points: [
                'Responsible for pair programming / mentoring and unblocking reports.',
                'Delegation and prioritization of tasks with a workload of about 25-45 tickets per week.',
                'React/Redux App on frontend for Admin and Checkout.',
                'Hybrid PHP / Go backend.',
                '4 years on the team working to deliver features and bug fixes for both the frontend and backend.'
            ],
            dataAos: 'fade-right',
            dataAosDuration: '1000',
            dataAosEasing: 'ease-in-out',
            dataAosOnce: 'true'
        },
        {
            className: 'Projects_Staples',
            name: 'Staples Project',
            role: 'Developer Team Lead / Full Stack Developer',
            points: [
                'React/Redux App.',
                'Customized the current functionality of Cashier to allow for Staples business requirements to be met.',
                'Successful delivery on tight timeline.'
            ],
            dataAos: 'fade-up',
            dataAosDuration: '1000',
            dataAosEasing: 'ease-in-out',
            dataAosOnce: 'true'
        },
        {
            className: 'Projects_MultiCurrency',
            name: 'Bold MultiCurrency',
            role: 'Completed the first iteration of the frontend of the app in its entirety.',
            points: [
                'React/Redux App converting prices in a Shopify / BigCommerce store, by currency and ruleset define by the merchant.',
                'Customized the current functionality of Cashier to allow for Staples business requirements to be met.',
                'Helped write the integration for Cashier to allow for charging in specific currencies.'
            ],
            dataAos: 'fade-up',
            dataAosDuration: '1000',
            dataAosEasing: 'ease-in-out',
            dataAosOnce: 'true'
        },
        {
            className: 'Projects_DeepRoots',
            name: 'Deep Roots Strong Dance - Website',
            role: 'Designed and built entire site',
            points: [
                'React/Redux App.',
                'API endpoint for adding to mailing list.',
            ],
            dataAos: 'fade-left',
            dataAosDuration: '1000',
            dataAosEasing: 'ease-in-out',
            dataAosOnce: 'true'
        }
    ];

    return (
        <div className="TechSummaryPage">
            <InfoSection
                data-aos="fade-left"
                data-aos-duration="1000"
                data-aos-easing="ease-in-out"
                data-aos-once="true"
                classnames="TechSummaryPage__Introduction">
                <div>
                    <h2>A Technical Brief History</h2>
                    <p>David has been intrigued with Technology from a very small age. He was first introduced by his parents via a IBM PS/2 486 Desktop PC.</p>
                    <p>David proceeded to brick the Windows installation within a week, pretty sure it was <code>FORMAT c:</code>.</p>
                    <p>When David's parents brought the PC to get fixed, he asked the person fixing the PC if he could watch.</p>
                    <p>After watching what he did to re-install Windows, David was now equipped to corrupt the OS as much as he pleased as long as he reinstalled it.</p>
                    <p>Little did David know that this interaction would spark a life long journey into technology.</p>
                </div>
                <div className="TechSummaryPage__Introduction__StatBars"></div>
            </InfoSection>
            <InfoSection classnames="TechSummaryPage__Projects">
                <h2>Projects</h2>
                {projects.map(project => <Project key={JSON.stringify(project)} {...project} />)}
            </InfoSection>
        </div>
    );
}

export default TechSummary;