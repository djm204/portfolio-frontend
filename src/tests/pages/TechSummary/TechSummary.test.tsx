import React from 'react'
import TechSummary from '../../../pages/TechSummary/TechSummary'
import { render } from '@testing-library/react'
import { HashRouter as Router } from 'react-router-dom'

it('TechSummary renders without crashing', () => {
  const result = render(
    <Router>
      <TechSummary />
    </Router>
  )

  expect(result).toMatchSnapshot()
})
