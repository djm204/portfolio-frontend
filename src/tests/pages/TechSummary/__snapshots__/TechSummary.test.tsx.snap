// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechSummary renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="TechSummaryPage"
        data-aos="fade-left"
        data-aos-duration="1000"
        data-aos-easing="ease-in-out"
        data-aos-once="true"
        id="TechSummary"
      >
        <div
          class="HorizontalSeparator lighter left-to-right"
          classnames="lighter,left-to-right"
        />
        <div
          class="InfoSection TechSummaryPage__InfoSection"
          classnames="TechSummaryPage__InfoSection"
        >
          <div
            class="TechSummaryPage__Introduction"
          >
            <h2>
              A Brief History
            </h2>
            <p>
              Technology has had me enamoured from a very young age. When I had a remote control car, I didn't just play with it, I took it apart to see what made the insides work. As you would imagine it didn't end well for young David.
            </p>
            <p>
              I was first introduced to computers by my parents via an IBM PS/2 486 desktop PC. I managed to mess up the Windows installation within a week. Pretty darn sure I used
               
              <code>
                FORMAT C:
              </code>
            </p>
            <p>
              When my parents brought the PC to get fixed, I asked the person fixing the PC if I could watch.
            </p>
            <p>
              After watching what he did to re-install Windows, I was now equipped to experiment with the OS as much as I pleased as long as I reinstalled it, of course.
            </p>
            <p>
              Little did I know that this interaction would spark a life long journey into technology.
            </p>
          </div>
          <div
            class="TechSummaryPage__Introduction__StatBars"
          >
            <div
              class="StatsBar__Wrapper"
            >
              <div
                class="StatsBar__Name"
              >
                HTML
              </div>
              <div
                class="StatsBar__Container"
              >
                <div
                  class="StatsBar"
                  style="width: 85%;"
                >
                  95%
                </div>
              </div>
            </div>
            <div
              class="StatsBar__Wrapper"
            >
              <div
                class="StatsBar__Name"
              >
                CSS
              </div>
              <div
                class="StatsBar__Container"
              >
                <div
                  class="StatsBar"
                  style="width: 80%;"
                >
                  90%
                </div>
              </div>
            </div>
            <div
              class="StatsBar__Wrapper"
            >
              <div
                class="StatsBar__Name"
              >
                JavaScript
              </div>
              <div
                class="StatsBar__Container"
              >
                <div
                  class="StatsBar"
                  style="width: 80%;"
                >
                  90%
                </div>
              </div>
            </div>
            <div
              class="StatsBar__Wrapper"
            >
              <div
                class="StatsBar__Name"
              >
                React
              </div>
              <div
                class="StatsBar__Container"
              >
                <div
                  class="StatsBar"
                  style="width: 80%;"
                >
                  90%
                </div>
              </div>
            </div>
            <div
              class="StatsBar__Wrapper"
            >
              <div
                class="StatsBar__Name"
              >
                Redux
              </div>
              <div
                class="StatsBar__Container"
              >
                <div
                  class="StatsBar"
                  style="width: 75%;"
                >
                  85%
                </div>
              </div>
            </div>
            <div
              class="StatsBar__Wrapper"
            >
              <div
                class="StatsBar__Name"
              >
                TypeScript
              </div>
              <div
                class="StatsBar__Container"
              >
                <div
                  class="StatsBar"
                  style="width: 65%;"
                >
                  75%
                </div>
              </div>
            </div>
            <div
              class="StatsBar__Wrapper"
            >
              <div
                class="StatsBar__Name"
              >
                NodeJS
              </div>
              <div
                class="StatsBar__Container"
              >
                <div
                  class="StatsBar"
                  style="width: 55%;"
                >
                  65%
                </div>
              </div>
            </div>
            <div
              class="StatsBar__Wrapper"
            >
              <div
                class="StatsBar__Name"
              >
                Go
              </div>
              <div
                class="StatsBar__Container"
              >
                <div
                  class="StatsBar"
                  style="width: 55%;"
                >
                  65%
                </div>
              </div>
            </div>
            <div
              class="StatsBar__Wrapper"
            >
              <div
                class="StatsBar__Name"
              >
                K8s
              </div>
              <div
                class="StatsBar__Container"
              >
                <div
                  class="StatsBar"
                  style="width: 45%;"
                >
                  55%
                </div>
              </div>
            </div>
            <div
              class="StatsBar__Wrapper"
            >
              <div
                class="StatsBar__Name"
              >
                GCP
              </div>
              <div
                class="StatsBar__Container"
              >
                <div
                  class="StatsBar"
                  style="width: 45%;"
                >
                  55%
                </div>
              </div>
            </div>
            <small>
              * Percentages represent level of confidence when working with respective technologies
            </small>
          </div>
        </div>
        <div
          class="HorizontalSeparator lighter right-to-left"
          classnames="lighter,right-to-left"
        />
        <div
          class="InfoSection TechSummaryPage__Projects"
          classnames="TechSummaryPage__Projects"
          id="Projects"
        >
          <h2>
            Projects
          </h2>
          <div
            class="TechSummary__Projects__Wrapper"
          >
            <div
              class="Project Projects_Cashier"
              id="Projects_Cashier"
            >
              <div
                class="ExpandedProjectImageBackground"
              >
                <div
                  class="Project__Image__Wrapper__Expanded"
                >
                  <img
                    alt="Cashier"
                    class="Project__Image__Expanded"
                    src="cashier--expanded.jpg"
                  />
                </div>
              </div>
              <img
                alt="Cashier"
                class="Project__Image"
                src="cashier.jpg"
              />
              <div
                class="Project__Details"
              >
                <h3
                  class="Project__Name "
                  data-aos-duration="1000"
                  data-aos-easing="ease-in-out"
                >
                  Cashier
                </h3>
                <div
                  class="Project__Role "
                >
                  Developer Team Lead / Full Stack Developer
                </div>
                <ul
                  class="Project__Notables "
                >
                  <li>
                    Pair programming / mentoring & unblocking reports.
                  </li>
                  <li>
                    Delegation and prioritization of tasks.
                  </li>
                  <li>
                    React/Redux App for Admin and Checkout.
                  </li>
                  <li>
                    Hybrid PHP / Go backend.
                  </li>
                  <li>
                    4 years on the team.
                  </li>
                </ul>
                <div
                  class="Project__Link__Wrapper"
                >
                  <span
                    class="Project__NotPublic "
                  >
                    *Not Publicly Available
                  </span>
                </div>
              </div>
            </div>
            <div
              class="Project Projects_Staples"
              id="Projects_Staples"
            >
              <div
                class="ExpandedProjectImageBackground"
              >
                <div
                  class="Project__Image__Wrapper__Expanded"
                >
                  <img
                    alt="Staples Project"
                    class="Project__Image__Expanded"
                    src="staples--expanded.jpg"
                  />
                </div>
              </div>
              <img
                alt="Staples Project"
                class="Project__Image"
                src="staples.jpg"
              />
              <div
                class="Project__Details"
              >
                <h3
                  class="Project__Name "
                  data-aos-duration="1000"
                  data-aos-easing="ease-in-out"
                >
                  Staples Project
                </h3>
                <div
                  class="Project__Role "
                >
                  Developer Team Lead / Full Stack Developer
                </div>
                <ul
                  class="Project__Notables "
                >
                  <li>
                    React/Redux App.
                  </li>
                  <li>
                    Customized Cashier to meet Staples' business requirements.
                  </li>
                  <li>
                    Successful delivery on tight timeline.
                  </li>
                </ul>
                <div
                  class="Project__Link__Wrapper"
                >
                  <a
                    class="Project__Link"
                    href="https://staples.ca"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    View Project
                  </a>
                </div>
              </div>
            </div>
            <div
              class="Project Projects_MultiCurrency"
              id="Projects_MultiCurrency"
            >
              <div
                class="ExpandedProjectImageBackground"
              >
                <div
                  class="Project__Image__Wrapper__Expanded"
                >
                  <img
                    alt="Bold MultiCurrency"
                    class="Project__Image__Expanded"
                    src="multicurrency--expanded.jpg"
                  />
                </div>
              </div>
              <img
                alt="Bold MultiCurrency"
                class="Project__Image"
                src="multicurrency.jpg"
              />
              <div
                class="Project__Details"
              >
                <h3
                  class="Project__Name "
                  data-aos-duration="1000"
                  data-aos-easing="ease-in-out"
                >
                  Bold MultiCurrency
                </h3>
                <div
                  class="Project__Role "
                >
                  Completed the first iteration of the frontend of the app in its entirety, pretty much solo. *cracks knuckles*
                </div>
                <ul
                  class="Project__Notables "
                >
                  <li>
                    React/Redux App
                  </li>
                  <li>
                    Conversion engine
                  </li>
                  <li>
                    Integrated with Cashier
                  </li>
                </ul>
                <div
                  class="Project__Link__Wrapper"
                >
                  <a
                    class="Project__Link"
                    href="https://boldcommerce.com/multi-currency"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    View Project
                  </a>
                </div>
              </div>
            </div>
            <div
              class="Project Projects_DeepRoots"
              id="Projects_DeepRoots"
            >
              <div
                class="ExpandedProjectImageBackground"
              >
                <div
                  class="Project__Image__Wrapper__Expanded"
                >
                  <img
                    alt="DRSD - Website"
                    class="Project__Image__Expanded"
                    src="deeproots--expanded.jpg"
                  />
                </div>
              </div>
              <img
                alt="DRSD - Website"
                class="Project__Image"
                src="deeproots.jpg"
              />
              <div
                class="Project__Details"
              >
                <h3
                  class="Project__Name "
                  data-aos-duration="1000"
                  data-aos-easing="ease-in-out"
                >
                  DRSD - Website
                </h3>
                <div
                  class="Project__Role "
                >
                  Designed and built entire site
                </div>
                <ul
                  class="Project__Notables "
                >
                  <li>
                    React/Redux
                  </li>
                  <li>
                    API endpoint for adding to mailing list.
                  </li>
                </ul>
                <div
                  class="Project__Link__Wrapper"
                >
                  <a
                    class="Project__Link"
                    href="https://www.deeproots-strongdance.com/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    View Project
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="TechSummaryPage"
      data-aos="fade-left"
      data-aos-duration="1000"
      data-aos-easing="ease-in-out"
      data-aos-once="true"
      id="TechSummary"
    >
      <div
        class="HorizontalSeparator lighter left-to-right"
        classnames="lighter,left-to-right"
      />
      <div
        class="InfoSection TechSummaryPage__InfoSection"
        classnames="TechSummaryPage__InfoSection"
      >
        <div
          class="TechSummaryPage__Introduction"
        >
          <h2>
            A Brief History
          </h2>
          <p>
            Technology has had me enamoured from a very young age. When I had a remote control car, I didn't just play with it, I took it apart to see what made the insides work. As you would imagine it didn't end well for young David.
          </p>
          <p>
            I was first introduced to computers by my parents via an IBM PS/2 486 desktop PC. I managed to mess up the Windows installation within a week. Pretty darn sure I used
             
            <code>
              FORMAT C:
            </code>
          </p>
          <p>
            When my parents brought the PC to get fixed, I asked the person fixing the PC if I could watch.
          </p>
          <p>
            After watching what he did to re-install Windows, I was now equipped to experiment with the OS as much as I pleased as long as I reinstalled it, of course.
          </p>
          <p>
            Little did I know that this interaction would spark a life long journey into technology.
          </p>
        </div>
        <div
          class="TechSummaryPage__Introduction__StatBars"
        >
          <div
            class="StatsBar__Wrapper"
          >
            <div
              class="StatsBar__Name"
            >
              HTML
            </div>
            <div
              class="StatsBar__Container"
            >
              <div
                class="StatsBar"
                style="width: 85%;"
              >
                95%
              </div>
            </div>
          </div>
          <div
            class="StatsBar__Wrapper"
          >
            <div
              class="StatsBar__Name"
            >
              CSS
            </div>
            <div
              class="StatsBar__Container"
            >
              <div
                class="StatsBar"
                style="width: 80%;"
              >
                90%
              </div>
            </div>
          </div>
          <div
            class="StatsBar__Wrapper"
          >
            <div
              class="StatsBar__Name"
            >
              JavaScript
            </div>
            <div
              class="StatsBar__Container"
            >
              <div
                class="StatsBar"
                style="width: 80%;"
              >
                90%
              </div>
            </div>
          </div>
          <div
            class="StatsBar__Wrapper"
          >
            <div
              class="StatsBar__Name"
            >
              React
            </div>
            <div
              class="StatsBar__Container"
            >
              <div
                class="StatsBar"
                style="width: 80%;"
              >
                90%
              </div>
            </div>
          </div>
          <div
            class="StatsBar__Wrapper"
          >
            <div
              class="StatsBar__Name"
            >
              Redux
            </div>
            <div
              class="StatsBar__Container"
            >
              <div
                class="StatsBar"
                style="width: 75%;"
              >
                85%
              </div>
            </div>
          </div>
          <div
            class="StatsBar__Wrapper"
          >
            <div
              class="StatsBar__Name"
            >
              TypeScript
            </div>
            <div
              class="StatsBar__Container"
            >
              <div
                class="StatsBar"
                style="width: 65%;"
              >
                75%
              </div>
            </div>
          </div>
          <div
            class="StatsBar__Wrapper"
          >
            <div
              class="StatsBar__Name"
            >
              NodeJS
            </div>
            <div
              class="StatsBar__Container"
            >
              <div
                class="StatsBar"
                style="width: 55%;"
              >
                65%
              </div>
            </div>
          </div>
          <div
            class="StatsBar__Wrapper"
          >
            <div
              class="StatsBar__Name"
            >
              Go
            </div>
            <div
              class="StatsBar__Container"
            >
              <div
                class="StatsBar"
                style="width: 55%;"
              >
                65%
              </div>
            </div>
          </div>
          <div
            class="StatsBar__Wrapper"
          >
            <div
              class="StatsBar__Name"
            >
              K8s
            </div>
            <div
              class="StatsBar__Container"
            >
              <div
                class="StatsBar"
                style="width: 45%;"
              >
                55%
              </div>
            </div>
          </div>
          <div
            class="StatsBar__Wrapper"
          >
            <div
              class="StatsBar__Name"
            >
              GCP
            </div>
            <div
              class="StatsBar__Container"
            >
              <div
                class="StatsBar"
                style="width: 45%;"
              >
                55%
              </div>
            </div>
          </div>
          <small>
            * Percentages represent level of confidence when working with respective technologies
          </small>
        </div>
      </div>
      <div
        class="HorizontalSeparator lighter right-to-left"
        classnames="lighter,right-to-left"
      />
      <div
        class="InfoSection TechSummaryPage__Projects"
        classnames="TechSummaryPage__Projects"
        id="Projects"
      >
        <h2>
          Projects
        </h2>
        <div
          class="TechSummary__Projects__Wrapper"
        >
          <div
            class="Project Projects_Cashier"
            id="Projects_Cashier"
          >
            <div
              class="ExpandedProjectImageBackground"
            >
              <div
                class="Project__Image__Wrapper__Expanded"
              >
                <img
                  alt="Cashier"
                  class="Project__Image__Expanded"
                  src="cashier--expanded.jpg"
                />
              </div>
            </div>
            <img
              alt="Cashier"
              class="Project__Image"
              src="cashier.jpg"
            />
            <div
              class="Project__Details"
            >
              <h3
                class="Project__Name "
                data-aos-duration="1000"
                data-aos-easing="ease-in-out"
              >
                Cashier
              </h3>
              <div
                class="Project__Role "
              >
                Developer Team Lead / Full Stack Developer
              </div>
              <ul
                class="Project__Notables "
              >
                <li>
                  Pair programming / mentoring & unblocking reports.
                </li>
                <li>
                  Delegation and prioritization of tasks.
                </li>
                <li>
                  React/Redux App for Admin and Checkout.
                </li>
                <li>
                  Hybrid PHP / Go backend.
                </li>
                <li>
                  4 years on the team.
                </li>
              </ul>
              <div
                class="Project__Link__Wrapper"
              >
                <span
                  class="Project__NotPublic "
                >
                  *Not Publicly Available
                </span>
              </div>
            </div>
          </div>
          <div
            class="Project Projects_Staples"
            id="Projects_Staples"
          >
            <div
              class="ExpandedProjectImageBackground"
            >
              <div
                class="Project__Image__Wrapper__Expanded"
              >
                <img
                  alt="Staples Project"
                  class="Project__Image__Expanded"
                  src="staples--expanded.jpg"
                />
              </div>
            </div>
            <img
              alt="Staples Project"
              class="Project__Image"
              src="staples.jpg"
            />
            <div
              class="Project__Details"
            >
              <h3
                class="Project__Name "
                data-aos-duration="1000"
                data-aos-easing="ease-in-out"
              >
                Staples Project
              </h3>
              <div
                class="Project__Role "
              >
                Developer Team Lead / Full Stack Developer
              </div>
              <ul
                class="Project__Notables "
              >
                <li>
                  React/Redux App.
                </li>
                <li>
                  Customized Cashier to meet Staples' business requirements.
                </li>
                <li>
                  Successful delivery on tight timeline.
                </li>
              </ul>
              <div
                class="Project__Link__Wrapper"
              >
                <a
                  class="Project__Link"
                  href="https://staples.ca"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  View Project
                </a>
              </div>
            </div>
          </div>
          <div
            class="Project Projects_MultiCurrency"
            id="Projects_MultiCurrency"
          >
            <div
              class="ExpandedProjectImageBackground"
            >
              <div
                class="Project__Image__Wrapper__Expanded"
              >
                <img
                  alt="Bold MultiCurrency"
                  class="Project__Image__Expanded"
                  src="multicurrency--expanded.jpg"
                />
              </div>
            </div>
            <img
              alt="Bold MultiCurrency"
              class="Project__Image"
              src="multicurrency.jpg"
            />
            <div
              class="Project__Details"
            >
              <h3
                class="Project__Name "
                data-aos-duration="1000"
                data-aos-easing="ease-in-out"
              >
                Bold MultiCurrency
              </h3>
              <div
                class="Project__Role "
              >
                Completed the first iteration of the frontend of the app in its entirety, pretty much solo. *cracks knuckles*
              </div>
              <ul
                class="Project__Notables "
              >
                <li>
                  React/Redux App
                </li>
                <li>
                  Conversion engine
                </li>
                <li>
                  Integrated with Cashier
                </li>
              </ul>
              <div
                class="Project__Link__Wrapper"
              >
                <a
                  class="Project__Link"
                  href="https://boldcommerce.com/multi-currency"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  View Project
                </a>
              </div>
            </div>
          </div>
          <div
            class="Project Projects_DeepRoots"
            id="Projects_DeepRoots"
          >
            <div
              class="ExpandedProjectImageBackground"
            >
              <div
                class="Project__Image__Wrapper__Expanded"
              >
                <img
                  alt="DRSD - Website"
                  class="Project__Image__Expanded"
                  src="deeproots--expanded.jpg"
                />
              </div>
            </div>
            <img
              alt="DRSD - Website"
              class="Project__Image"
              src="deeproots.jpg"
            />
            <div
              class="Project__Details"
            >
              <h3
                class="Project__Name "
                data-aos-duration="1000"
                data-aos-easing="ease-in-out"
              >
                DRSD - Website
              </h3>
              <div
                class="Project__Role "
              >
                Designed and built entire site
              </div>
              <ul
                class="Project__Notables "
              >
                <li>
                  React/Redux
                </li>
                <li>
                  API endpoint for adding to mailing list.
                </li>
              </ul>
              <div
                class="Project__Link__Wrapper"
              >
                <a
                  class="Project__Link"
                  href="https://www.deeproots-strongdance.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  View Project
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
